bundles/binding/org.openhab.binding.weather/src/main/java/org/openhab/binding/weather/internal/common/WeatherConfig.java

```diff
@@ -61,62 +61,77 @@
+        }
+
+        Enumeration<String> keys = properties.keys();
+        while (keys.hasMoreElements()) {
+            String key = keys.nextElement();
+
+            String value = StringUtils.trimToNull((String) properties.get(key));
+            if (StringUtils.startsWithIgnoreCase(key, "apikey")) {
+                parseApiKey(key, value);
+            } else if (StringUtils.startsWithIgnoreCase(key, "location")) {
+                parseLocation(key, value);
             }
+        }
 
-            // check all LocationConfigs
-            for (LocationConfig lc : locationConfigs.values()) {
-                if (!lc.isValid()) {
-                    throw new ConfigurationException("weather", "Incomplete location config for locationId '"
-                            + lc.getLocationId() + "', please check your openhab.cfg!");
-                }
-                if (lc.getProviderName() != ProviderName.YAHOO && !providerConfigs.containsKey(lc.getProviderName())) {
-                    throw new ConfigurationException("weather", "No apikey found for provider '" + lc.getProviderName()
-                            + "', please check your openhab.cfg!");
-                }
+        // check all LocationConfigs
+        for (LocationConfig lc : locationConfigs.values()) {
+            if (!lc.isValid()) {
+                parseCompleted = true;
+                logger.warn("Incomplete location config for locationId '{}'. Check openhab.cfg.", lc.getLocationId());
+                throw new ConfigurationException("weather", "Incomplete location config for locationId '"
+                        + lc.getLocationId() + "'. Check openhab.cfg.");
             }
 
-            // check all ProviderConfigs
-            for (ProviderConfig pc : providerConfigs.values()) {
-                if (!pc.isValid()) {
-                    throw new ConfigurationException("weather", "Invalid apikey config for provider '"
-                            + pc.getProviderName() + "', please check your openhab.cfg!");
-                }
+            if (lc.getProviderName() != ProviderName.YAHOO && !providerConfigs.containsKey(lc.getProviderName())) {
+                parseCompleted = true;
+                logger.warn("No apikey found for provider '{}'. Check openhab.cfg.", lc.getProviderName());
+                throw new ConfigurationException("weather", "No apikey found for provider '" + lc.getProviderName()
+                        + "'. Check openhab.cfg.");
             }
+        }
 
-            valid = locationConfigs.size() > 0;
+        // check all ProviderConfigs
+        for (ProviderConfig pc : providerConfigs.values()) {
+            if (!pc.isValid()) {
+                parseCompleted = true;
+                logger.warn("Invalid apikey config for provider '{}'. Check openhab.cfg.", pc.getProviderName());
+                throw new ConfigurationException("weather", "Invalid apikey config for provider '"
```

Comment:
As far as I can see, yes. I haven't changed it, though, so it should still be as accurate as it has ever been.